mne/viz/utils.py

```diff
@@ -145,9 +145,19 @@ def plt_show(show=True, fig=None, **kwargs):
 
     if hasattr(fig, "mne") and hasattr(fig.mne, "backend"):
         backend = fig.mne.backend
+        # TODO: This is a hack to deal with the fact that the
+        # with plt.ion():
+        #     BACKEND = get_backend()
+        # an the top of _mpl_figure detects QtAgg during testing even though
+        # we've set the backend to Agg.
+        if backend != "agg":
+            gotten_backend = get_backend()
+            if gotten_backend == "agg":
+                backend = "agg"
```

Comment:
FYI @drammock the global `BACKEND` business seems to be the cause of needing 
```
    # TODO: This is indicative of a problem
    ignore:.*Matplotlib is currently using agg.*:
```
in conftest... I'm trying a hopefully better workaround here.